modules/table/src/test/java/org/apache/ignite/internal/table/distributed/TableUtilsTest.java

```diff
@@ -35,95 +37,163 @@
+        String indexName2 = INDEX_NAME + 2;
+        String indexName3 = INDEX_NAME + 3;
+        String indexName4 = INDEX_NAME + 4;
 
-        try {
-            consumer.accept(catalogManager);
-        } finally {
-            closeAll(catalogManager::beforeNodeStop, catalogManager::stop);
+        for (String indexName : List.of(indexName0, indexName1, indexName2, indexName3, indexName4)) {
+            createSimpleHashIndex(catalogManager, TABLE_NAME, indexName);
         }
+
+        int indexId0 = indexId(indexName0);
+        int indexId1 = indexId(indexName1);
+        int indexId2 = indexId(indexName2);
+        int indexId3 = indexId(indexName3);
+        int indexId4 = indexId(indexName4);
+
+        for (String indexName : List.of(indexName1, indexName2, indexName3, indexName4)) {
+            startBuildingIndex(catalogManager, indexId(indexName));
+        }
+
+        for (String indexName : List.of(indexName2, indexName3, indexName4)) {
+            makeIndexAvailable(catalogManager, indexId(indexName));
+        }
+
+        for (String indexName : List.of(indexName3, indexName4)) {
+            dropIndex(catalogManager, DEFAULT_SCHEMA_NAME, indexName);
+        }
+
+        removeIndex(catalogManager, indexId4);
+
+        CatalogManager spy = spy(catalogManager);
+
+        HybridTimestamp beginTs = clock.now();
+
+        int tableId = getTableIdStrict(catalogManager, TABLE_NAME, clock.nowLong());
+
+        assertThat(
+                indexIdsAtRwTxBeginTs(spy, transactionId(beginTs, 1), tableId),
+                contains(
+                        indexId(pkIndexName(TABLE_NAME)),
+                        indexId0,
+                        indexId1,
+                        indexId2,
+                        indexId3
+                )
+        );
+
+        verify(spy).activeCatalogVersion(eq(beginTs.longValue()));
+        verify(spy).indexes(eq(catalogManager.activeCatalogVersion(beginTs.longValue())), eq(tableId));
+    }
+
+    @Test
+    void testDroppedTables() {
+        String tableName0 = TABLE_NAME + 0;
+        String tableName1 = TABLE_NAME + 1;
+        String tableName2 = TABLE_NAME + 2;
+
+        createSimpleTable(catalogManager, tableName0);
+        createSimpleTable(catalogManager, tableName1);
+        createSimpleTable(catalogManager, tableName2);
+
+        int tableId0 = tableId(tableName0);
+        int tableId1 = tableId(tableName1);
+
+        int catalogVersionBeforeRemoveTable1 = catalogManager.latestCatalogVersion();
+        dropSimpleTable(catalogManager, tableName1);
+
+        int catalogVersionBeforeRemoveTable0 = catalogManager.latestCatalogVersion();
+        dropSimpleTable(catalogManager, tableName0);
+
+        assertThat(droppedTables(null), empty());
+        assertThat(droppedTables(catalogTime(catalogVersionBeforeRemoveTable1)), empty());
+        assertThat(droppedTables(catalogTime(catalogVersionBeforeRemoveTable1).addPhysicalTime(1)), empty());
```

Comment:
What does this statement test?

Reply:
It checks that not only the directory version is working correctly for the duration of activation, but also for a slightly longer period of time. I can remove the check if you wish.

Reply:
Please add a comment

Reply:
I tried to add a comment.